toolchain/check/impl_lookup.cpp

```diff
@@ -548,7 +548,24 @@ static auto TypeCanAggregateDestroy(Context& context,
     -> bool {
   auto inst = context.insts().Get(
       context.constant_values().GetInstId(query_self_const_id));
-  return inst.Is<SemIR::StructType>() || inst.Is<SemIR::TupleType>();
+  CARBON_KIND_SWITCH(inst) {
+    case CARBON_KIND(SemIR::ClassType class_type): {
+      // Carbon classes will generate a `Destroy` impl, but we use this to
+      // provide destruction for `Cpp`-scoped classes.
+      // TODO: Don't provide this for C++ types that lack a destructor.
+      auto class_info = context.classes().Get(class_type.class_id);
+      return class_info.is_complete() && context.name_scopes()
```

Comment:
Hard to validate myself - but is the incomplete case covered by tests?

Reply:
No, not covered. I think it may be needed, but removed the validation because there's a TODO anyways.